clang/lib/Sema/CheckExprLifetime.cpp

```diff
@@ -249,7 +254,7 @@ static void visitLocalsRetainedByReferenceBinding(IndirectLocalPath &Path,
                                                   LocalVisitor Visit);
 
 template <typename T> static bool isRecordWithAttr(QualType Type) {
-  if (auto *RD = Type->getAsCXXRecordDecl())
+  if (auto *RD = Type.getNonReferenceType()->getAsCXXRecordDecl())
```

Comment:
Removing this will create a false positive for the following 2 cases only:
```cpp
vsv.push_back(getOptionalSV().value());
vsv.push_back(getOptionalMySV().value());
```

Reply:
I will remove this for now from this PR and leave the tests with a FIXME. Probably needs a separate PR to keep this PR simpler.